Presentation: Learning snippets cleanup (#3386)
iTwin/itwinjs-core: docs/presentation/Advanced/Testing.md

# Testing

Although the presentation framework itself is thoroughly tested, consumers should still verify they get expected results for their iModel + ruleset combinations. The **@itwin/presentation-testing** package is delivered
purely for that reason.

The package delivers an API that allows creating hierarchies for supplied iModels and rulesets. Consumers can then verify the result using tools of their liking. Our recommendation is to use snapshot testing for 2 reasons:

> 1. Resulting hierarchies get rather large - testing the them in code might be difficult.

Reviewer: ```suggestion
1. Resulting hierarchies get rather large — testing them in code might be difficult.
```

2. Snapshots protect against regressions.

## Example

An example of setting up snapshot tests with the **@itwin/presentation-testing** package:

1. Initialize testing library and open iModel

    ```ts
    [[include:Presentation.Testing.Rulesets.Setup]]
    ```

2. Generate and verify results:

    2.1. Hierarchies:

      ```ts
      [[include:Presentation.Testing.Rulesets.Hierarchies]]
      ```

    2.2. Content:

      ```ts
      [[include:Presentation.Testing.Rulesets.Content]]
      ```

3. Close iModel and terminate testing library:

    ```ts
    [[include:Presentation.Testing.Rulesets.Terminate]]
    ```

> **Note:** The above example uses [mocha](https://www.npmjs.com/package/mocha) and [chai-jest-snapshot](https://www.npmjs.com/package/chai-jest-snapshot) packages to perform snapshot testing. The path returned by `createSnapshotPath` function used in the tests should match the project structure and is outside the scope of this example.

Reviewer: > The path returned by `createSnapshotPath` function used in the tests should match the project structure [..]

Does chai-jest-snapshot really care where you place snapshot files? I think you should omit this sentence.

Author: I'm using this function in the example code and now including its content. I think it makes sense to tell what it's for.

Reviewer: Does chai-jest-snapshot really care where you place snapshot files though? I find this sentence potentially misleading.

Reviewer: Then let's not mislead people that a particular file structure is required for snapshots to work... What `createSnapshotPath` does is obvious from the name, but if you insist on over explaining it, you can do it without imposing our internal requirements to the end users:

```suggestion
**Note:** The above example uses [mocha](https://www.npmjs.com/package/mocha) and [chai-jest-snapshot](https://www.npmjs.com/package/chai-jest-snapshot) packages to perform snapshot testing. `createSnapshotPath` function retrieves a path to the snapshot for the current test.
```
